stakingModel/contracts/UBIScheme.sol

```diff
@@ -292,12 +286,19 @@ contract UBIScheme is AbstractUBI {
         returns (bool)
     {
         uint256 claimers = 0;
-        for(uint256 i = 0; i < accounts.length && i < nextDayDistributionCount[currentDay]; ++i) {
-            if (nextDayDistributionAddresses[currentDay][accounts[i]].isEntitled &&
-                !nextDayDistributionAddresses[currentDay][accounts[i]].isCompleted) { // avoiding of double claiming
-                nextDayDistributionAddresses[currentDay][accounts[i]].isCompleted = true;
-                require(_claim(accounts[i]), "distributed claim was failed");
-                claimers = claimers.add(1);
+        for(uint256 i = 0; i < accounts.length && i < activeUsersCount; ++i) {
+            if(pendingUsersAddresses[accounts[i]].isPending) {
+                if(identity.isWhitelisted(accounts[i])) {
+                    if(isActiveUser((accounts[i]))) { // avoids exception
```

Comment:
i think all these logic should be in _claim
claim should not throw any exceptions. i can return true/false

Reply:
moved to claim. no exceptions. only true / false and events
3a130df16ce6b41f40192c76921b58e63d2c5ccd